src/graph.rs

```diff
@@ -288,6 +288,10 @@ impl Attributes {
     {
         self.values.get(name.borrow())
     }
+
+    pub fn iter(&self) -> std::collections::hash_map::Iter<'_, Identifier, Value> {
```

Comment:
Same comment as in previous PR, you can use `impl Iterator<Item=&'_ Value> + '_` as the return type